### PR TITLE
Support auth check of redaction events in rooms >= v3

### PR DESCRIPTION
**Pull request by Cnly**
This also adds a `RoomVersion` field in `createContent`.

Signed-off-by: Alex Chen <minecnly@gmail.com>

### CODE REVIEW

File: eventauth.go [hunk 1]
```diff
@@ -632,7 +636,15 @@ func redactEventAllowed(event Event, authEvents AuthEventProvider) error {
 		return err
 	}
 
-	senderDomain, err := domainFromID(event.Sender())
+	roomVersion := allower.create.RoomVersion
+	if roomVersion != nil && *roomVersion != "1" && *roomVersion != "2" {
```
Comment (anoadragon453) [inline eventauth.go]: I assume we should start defining these as constants at some point, but that can wait until full room version support is implemented.

File: eventauth.go [hunk 2]
```diff
@@ -632,7 +636,15 @@ func redactEventAllowed(event Event, authEvents AuthEventProvider) error {
...
+		// We always accept redaction events into DAG for rooms >= v3 after the
```
Comment (anoadragon453) [inline eventauth.go]:
```suggestion
		// We always accept redaction events into the DAG for rooms >= v3 after the
```

File: eventcontent.go [hunk 3]
```diff
@@ -35,6 +35,8 @@ type createContent struct {
 	Federate *bool `json:"m.federate"`
 	// The creator of the room tells us what the default power levels are.
 	Creator string `json:"creator"`
+	// The version of the room. Should be treated as "1" when the key doesn't exist.
+	RoomVersion *string `json:"room_version"`
```
Comment (anoadragon453) [inline eventcontent.go]:
```suggestion
	RoomVersion *string `json:"room_version,omitempty"`
```

Comment (Cnly) [reply eventcontent.go]: Just a note that the `m.federate` key above is also optional. But `createContent` is not a public struct and has limited use so this should not be a big deal.

Comment (anoadragon453) [reply eventcontent.go]: I see, I suppose anything we send back to the client should have `"1"` in there anyways.

**Pull request merged**